java/code/src/com/suse/manager/webui/controllers/ProxyController.java

```diff
@@ -127,4 +94,39 @@ public ModelAndView containerConfig(Request requestIn, Response responseIn, User
         Map<String, Object> data = new HashMap<>();
         return new ModelAndView(data, "templates/proxy/container-config.jade");
     }
+
+    /**
+     * Create the proxy containers configuration.
+     *
+     * @param request the request object
+     * @param response the response object
+     * @param user the user
+     *
+     * @return the config file
+     */
+    public byte[] generateContainerConfig(Request request, Response response, User user) {
+        ProxyContainerConfigJson data = GSON.fromJson(request.body(),
+                new TypeToken<ProxyContainerConfigJson>() { }.getType());
+        if (!data.isValid()) {
+            LOG.error("Invalid input data");
+            halt(HttpStatus.SC_BAD_REQUEST);
+        }
+        try {
+            byte[] config = systemManager.createProxyContainerConfig(user, data.getProxyFqdn(),
+                    data.getProxyPort(), data.getServerFqdn(), data.getMaxCache(), data.getEmail(),
+                    data.getRootCA(), data.getIntermediateCAs(), data.getProxyCertPair(),
+                    data.getCaPair(), data.getCaPassword(), data.getCertData());
+
+            String serverName = data.getProxyFqdn().split("\\.")[0];
+            response.header("Content-Disposition", "attachment; filename=\"" + serverName + "-config.zip\"");
+            response.header("Content-Length", Integer.toString(config.length));
+            response.type("application/zip");
+            return config;
```

Comment:
Idea to discuss: do we want to return the `config.zip` directly and force the user to save it on the fly, or we generate it, we park it on the server, and we tell the user where it is and we also provide a download link just in case? I'm fine with both approaches, what do you think?